tests/PostcodeFormatterTest.php

```diff
@@ -76,7 +85,10 @@ public function providerFormat() : array
         return [
             ['GB', 'WC2E9RZ', 'WC2E 9RZ'],
             ['gb', 'wc-2E9RZ', 'WC2E 9RZ'],
-            ['PL', '12345', '12-345']
+            ['PL', '12345', '12-345'],
+            ['AT', 'A-8084', '8084'],
+            ['BE', 'B-1245', '1245'],
+            ['LU', 'L2556', '2556'],
```

Comment:
@BenMorel I just did that but there is some inconsistent behaviour in the tests, this happens because for the PostcodeFormatterTest.php file the global PostcodeFormatter class is used but for the individual tests it's not. This results in input not automatically being uppercased and also spaces not being removed etc. as happens for the other tests and also for the public API. 

In my opinion the abstract class CountryPostcodeFormatterTest should use the global PostcodeFormatter and then get the CountryCode via a constant or (there it is again :-) Reflection or something. This wil also test and ensure that classes are named properly and autoloaded correctly.

Reply:
Yeah sorry, I meant without the dashes:

```
['A8084', '8084'],
['B8084', null],
```

> In my opinion the abstract class CountryPostcodeFormatterTest should use the global PostcodeFormatter and then get the CountryCode via a constant or (there it is again :-) Reflection or something.

I understand where you're coming from. The idea is to test each country formatter in isolation (a unit test), without repeating the tests for "strip dashes and spaces" etc., which is the responsibility of the PostcodeFormatter tests.

> This wil also test and ensure that classes are named properly and autoloaded correctly.

That's a valid point. To improve this, we could:

- either add another test to each `CountryPostcodeFormatter`, that invokes the `PostcodeFormatter`
- or (preferred, IMO), ensure that each supported country has a test in `PostcodeFormatterTest` (I'm not sure if more than one test is needed here—we don't want to repeat the unit tests—but typically the one test would be dash- or space-separated as appropriate for the country)

What about discussing this in another PR? We already have 2 unrelated changes in the present one.

Reply:
Yes i get the idea, ofcourse unit tests are ment to test units, the only problem is that the stripe here is part of the format i'm trying to test and implement. And so i want it to be in the test somewhere. And also test that others are or different cases are still failing. 

Becasue what if you are changing the main class PostcodeFormatter, and remove the stripping of the stripes or spaces or add some other modification, then there is no way to test if this particular format still works. Or what if the removing of stripes is conflicting with one of the other formats.

But you are right let's indeed open a separate PR for this. For now i made it work with the latest commit.